### PR TITLE
Split cli from core

### PR DESCRIPTION
**Pull request by gabro**
This PR splits the `cli` project from `core`.

Now `core` only depends on `scalameta`. This is a first crucial step towards Scala.js compatibility ;)

The functionality is the same and the tests pass unchanged (although tests have been split too).

Also, I got rid of the old `metarpheus` package. Everything should be consistent now and use `io.buildo.metarpheus` as a root package.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -28,23 +28,32 @@ lazy val root = project.in(file("."))
...
+    commonSettings,
+    name := "metarpheus-cli",
+    libraryDependencies ++= Seq(
+      "org.rogach" %% "scallop" % "1.0.0",
+      "org.json4s" %% "json4s-jackson" % "3.5.0",
+      "com.twitter" %% "util-eval" % "6.41.0",
+      "org.scalatest" %% "scalatest" % "3.0.1" % Test
+    )
   )
+  .dependsOn(core)
```
Comment (bytecodeguru) [inline build.sbt]: Missing new line.

File: cli/src/main/scala/io.buildo.metarpheus/cli/Cli.scala [hunk 2]
```diff
@@ -69,7 +52,15 @@ object main {
...
+    if (target.isDirectory) {
+      val these: Array[java.io.File] = Option(target.listFiles).getOrElse(Array())
+      these ++ these.filter(_.isDirectory).flatMap(recursivelyListFiles)
+    } else {
+      Array(target)
+    }
   }
 
-}
+}
```
Comment (bytecodeguru) [inline cli/src/main/scala/io.buildo.metarpheus/cli/Cli.scala]: Missing new line.

File: core/src/main/scala/io.buildo.metarpheus/core/Metarpheus.scala [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+      parsed = parsed,
+      routeOverrides = config.routeOverrides,
+      routeMatcherToIntermediate = config.routeMatcherToIntermediate,
+      authRouteTermNames = config.authRouteTermNames,
+      wiro = wiro
+    )
+  }
+
+
+}
```
Comment (bytecodeguru) [inline core/src/main/scala/io.buildo.metarpheus/core/Metarpheus.scala]: Missing new line.

File: core/src/main/scala/io.buildo.metarpheus/core/extractors/package.scala [hunk 4]
```diff
@@ -124,4 +121,4 @@ package object extractors {
 
       (Some(desc.mkString(" ")), getTags(Nil, tagLines))
     }.getOrElse((None, List()))
-}
+}
```
Comment (bytecodeguru) [inline core/src/main/scala/io.buildo.metarpheus/core/extractors/package.scala]: MNL

File: core/src/test/scala/io.buildo.metarpheus/core/extractors/RouteSuite.scala [hunk 5]
```diff
@@ -1,25 +1,28 @@
...
 
+import extractors._
+
 class RouteSuite extends FunSuite {
   lazy val parsed = {
     import scala.meta._
-    import scala.meta.dialects.Scala211
-    morpheus.Fixtures.routes.parse[Source].get
+    val x = Fixtures.routes.parse[Source].get
+    x
```
Comment (bytecodeguru) [inline core/src/test/scala/io.buildo.metarpheus/core/extractors/RouteSuite.scala]: Why `val x` and then `x`?

Comment (gabro) [inline core/src/test/scala/io.buildo.metarpheus/core/extractors/RouteSuite.scala]: ops

**Pull request merged**